### PR TITLE
Support newline separated redirect_uris

### PR DESCRIPTION
**Pull request by matthewp**
Fixes https://github.com/superseriousbusiness/gotosocial/issues/962

### CODE REVIEW

File: internal/processing/app.go [hunk 1]
```diff
@@ -50,12 +51,21 @@ func (p *processor) AppCreate(ctx context.Context, authed *oauth.Auth, form *api
 		return nil, gtserror.NewErrorInternalError(err)
 	}
 
+	// Find the first real URI if there is one
+	var redirectURI string
```
Comment (tsmethurst) [inline internal/processing/app.go:55]: I'm curious about this: doesn't this mean that we basically throw away any other redirect URIs after the first non-oob one that we find?

**Comment by tsmethurst** [comment]
I'm not 100% sure about this one. It's nice to get client stuff working, but this seems like a workaround for a weird Mastodon API thingy. Does it conform to oauth2 spec? In general we're pretty flexible about Mastodon API quirks, but I don't want to do anything that breaks oauth2 spec conformance, so I would like to validate this first. Would you mind looking into it?

**Comment by matthewp** [comment]
@tsmethurst Yeah sure, but when you search for `redirect_uris` you only find references to Google APIs. I'm not sure it's an oauth thing.

Here's what makes sense to me though. When you register an app you can send it multiple redirect uris (maybe you have one for local testing, one for staging, one for prod).

Then when you go to do an authorization you send `redirect_uri` and the server makes sure it matches one of the ones registered for the app.

So perhaps what GoToSocial is doing wrong is it's using the value in the database to try to redirect to, rather than using the one that comes in the request. I'll look and see if that's something we can change.

**Comment by matthewp** [comment]
Ok, I have a setup locally where I can test against the client that is failing. This will help track down the problem better, will submit a new PR when I have a better idea of the problem.

**Comment by matthewp** [comment]
Ok, so I understand what is going on here now. Reopening the PR for discussion although it's not correct in its current form.

GTS saves the RedirectURIs as the Domain on the client here: https://github.com/superseriousbusiness/gotosocial/blob/a1dda2267260472aa777ef42d9d2f5c36df648f3/internal/processing/app.go#L73

When creating the auth code, the oauth library checks to see if the redirect uri is part of the client's domain. Since our client's domain is not a valid URI, it throws.

So *I think* the proper solution would be to create multiple clients, one for each redirect URI. But that would probably also mean making multiple Applications since those contain the client id / secret as well. And that doesn't feel right.

So that's the issue, would love to hear your thoughts @tsmethurst

**Comment by matthewp** [comment]
The client I was using is no longer sending redirect_uris. So I'm going to close this issue. I can move my last comment over to #962 so we can continue the discussion there.

**Pull request closed without merging**